petab/v2/lint.py

```diff
@@ -0,0 +1,549 @@
+            mapping_df=problem.mapping_df,
+        )
+
+        actual = set(problem.parameter_df.index)
+        missing = required - actual
+        extraneous = actual - allowed
+
+        # missing parameters might be present under a different name based on
+        # the mapping table
+        if missing and problem.mapping_df is not None:
+            model_to_petab_mapping = {}
+            for map_from, map_to in zip(
+                problem.mapping_df.index.values,
+                problem.mapping_df[MODEL_ENTITY_ID],
+                strict=True,
+            ):
+                if map_to in model_to_petab_mapping:
+                    model_to_petab_mapping[map_to].append(map_from)
+                else:
+                    model_to_petab_mapping[map_to] = [map_from]
+            missing = {
+                missing_id
+                for missing_id in missing
+                if missing_id not in model_to_petab_mapping
+                or all(
+                    mapping_parameter not in actual
+                    for mapping_parameter in model_to_petab_mapping[missing_id]
+                )
+            }
+
+        if missing:
+            return ValidationError(
+                "Missing parameter(s) in the model or the "
+                "parameters table: " + str(missing)
+            )
+
+        if extraneous:
+            return ValidationError(
+                "Extraneous parameter(s) in parameter table: "
+                + str(extraneous)
+            )
+
+
+class CheckValidParameterInConditionOrParameterTable(ValidationTask):
+    """A task to check that all required and only allowed model parameters are
+    present in the condition or parameter table."""
+
+    def run(self, problem: Problem) -> ValidationIssue | None:
+        if (
+            problem.model is None
+            or problem.condition_df is None
+            or problem.parameter_df is None
+        ):
+            return
+
+        try:
+            assert_model_parameters_in_condition_or_parameter_table(
+                problem.model,
+                problem.condition_df,
+                problem.parameter_df,
+                problem.mapping_df,
+            )
+        except AssertionError as e:
+            return ValidationIssue(
+                level=ValidationIssueSeverity.ERROR, message=str(e)
+            )
+
+
+class CheckVisualizationTable(ValidationTask):
+    """A task to validate the visualization table of a PEtab problem."""
+
+    def run(self, problem: Problem) -> ValidationIssue | None:
+        if problem.visualization_df is None:
+            return
+
+        from petab.visualize.lint import validate_visualization_df
+
+        if validate_visualization_df(problem):
+            return ValidationIssue(
+                level=ValidationIssueSeverity.ERROR,
+                message="Visualization table is invalid.",
+            )
+
+
+def get_required_parameters_for_parameter_table(
```

Comment:
Didn't check, let me know if I should

Reply:
The only change is the addition of ` and p not in problem.observable_df.index` in L513.